Library/OcAfterBootCompatLib/CustomSlide.c

```diff
@@ -89,11 +89,9 @@ GenerateSlideValue (
     return SlideSupport->ValidSlides[0];
   }
 
-  do {
-    DivU64x32Remainder (GetPseudoRandomNumber64 (), SlideSupport->ValidSlideCount, &Slide);
-  } while (Slide == 0);
+  DivU64x32Remainder (GetPseudoRandomNumber64 (), SlideSupport->ValidSlideCount - 1, &Slide);
```

Comment:
This change looks wrong to me, as now it avoids the first slide entry in ValidSlides, but it is not necessary 0 (e.g. on Z77 it normally starts with 12, 13, and so on). Please change it back to the original code, that is what we reverse-engineered from Efiboot.

Reply:
Or if the condition should be `SlideSupport->ValidSlides[Slide] == 0`  ?

Reply:
Good question, now I see the issue. Let me recheck EfiBoot.